src/Mod/Fem/femsolver/calculix/writer.py

```diff
@@ -909,6 +912,49 @@ def write_constraints_selfweight(self, f):
         # different element sets for different density
         # are written in the material element sets already
 
+    # ********************************************************************************************
+    # constraints centrif
+    def write_constraints_centrif(self, f):
+        if not self.centrif_objects:
+            return
+
+        # write constraint to file
+        f.write("\n***********************************************************\n")
+        f.write("** Centrif Constraint\n")
+        f.write("** written by {} function\n".format(sys._getframe().f_code.co_name))
+        for femobj in self.centrif_objects:
+            # femobj --> dict, FreeCAD document object is femobj["Object"]
+            centrif_obj = femobj["Object"]
```

Comment:
Can you please document the code a bit. When had a look at the FEM code it was hard to understand it due to lack of comments.
It is not necessary to comment what you do but why. So why is the CCX code as it is?

Reply:
generally this is true, but ... If a new object is added to FEM on a lot of places in the FEM code just some lines needs to be copied. I might should create some better comments in this regard. But my version of a good documentation is a good git history of new code added to FEM.